operators/pkg/examagent/instance.go

```diff
@@ -0,0 +1,289 @@
+		fmt.Fprintf(w, "Instance %s cannot be %s", instance.Name, op)
+		return
+	}
+
+	w.Header().Add("Content-Type", "application/json")
+
+	switch op {
+	case ctrlutil.OperationResultCreated:
+		w.WriteHeader(http.StatusCreated)
+	case ctrlutil.OperationResultUpdated:
+		w.WriteHeader(http.StatusOK)
+	case ctrlutil.OperationResultNone:
+		w.WriteHeader(http.StatusNotModified)
+	default:
+		w.WriteHeader(http.StatusInternalServerError)
+	}
+
+	if err := json.NewEncoder(w).Encode(AdapterFromInstance(instance)); err != nil {
+		fmt.Fprint(w, op)
+		log.Error(err, "operation complete but cannot encode instance")
+		return
+	}
+	log.Info("success")
+}
+
+// HandleGetAll handles the GET request for all the instances.
+func (ih *InstanceHandler) HandleGetAll(w http.ResponseWriter, r *http.Request, log logr.Logger) {
+	var instances clv1alpha2.InstanceList
+	// get all the instances in the namespace
+	if err := ih.Client.List(r.Context(), &instances, client.InNamespace(Options.TemplatesNS)); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprint(w, "Error retrieving instances")
+		log.Error(err, "error retrieving instances")
+		return
+	}
+
+	adapters := make([]InstanceAdapter, len(instances.Items))
+	for i := range instances.Items {
+		adapters[i] = *AdapterFromInstance(&instances.Items[i])
+	}
+
+	w.Header().Add("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(adapters); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprint(w, "Internal server error")
+		log.Error(err, "cannot encode instances")
+		return
+	}
+}
+
+// GetInstanceIDFromRequest returns the instance id from the request.
+func GetInstanceIDFromRequest(r *http.Request) string {
+	InstanceEP := path.Join(Options.BasePath, InstanceAdapterEndpoint) + "/"
+	instID := strings.Replace(r.URL.Path, InstanceEP, "", 1)
+	return instID
+}
+
+// HandleDelete handles the DELETE request for a InstanceAdapter api call.
+func (ih *InstanceHandler) HandleDelete(w http.ResponseWriter, r *http.Request, log logr.Logger) {
+	if err := Options.CheckAllowedIP(r.Header.Get(XForwardedFor)); err != nil {
+		log.Error(err, "unauthorized")
+		w.WriteHeader(http.StatusForbidden)
+		fmt.Fprint(w, "Forbidden")
+		return
+	}
+
+	inst := EmptyInstanceFromRequest(r)
+	log = log.WithValues("instance", inst.Name, "operation", "delete")
+	if err := ih.Client.Delete(r.Context(), inst); err != nil {
+		log.Error(err, "failed performing operation")
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprint(w, "Error deleting instance")
+		return
+	}
+
+	log.Info("success")
+}
+
+// EmptyInstanceFromRequest returns an Instance from a given request with just the ObjectMeta field set.
+func EmptyInstanceFromRequest(r *http.Request) *clv1alpha2.Instance {
+	instID := GetInstanceIDFromRequest(r)
+
+	inst := &clv1alpha2.Instance{
+		ObjectMeta: metav1.ObjectMeta{Name: instID, Namespace: Options.TemplatesNS},
+	}
+	return inst
```

Comment:
```suggestion
	return &clv1alpha2.Instance{
		ObjectMeta: metav1.ObjectMeta{Name: GetInstanceIDFromRequest(r), Namespace: Options.TemplatesNS},
	}
```

Nit. Just a bit shorter